api/metric/sdkhelpers.go

```diff
@@ -29,20 +55,35 @@ type LabelSetDelegate interface {
 	Delegate() LabelSet
 }
 
-// InstrumentImpl is the implementation-level interface Set/Add/Record
-// individual metrics without precomputed labels.
+// InstrumentImpl is a common interface for synchronous and
+// asynchronous instruments.
 type InstrumentImpl interface {
+	// Interface returns the underlying implementation of the
+	// instrument, which allows the implementation to gain access
+	// to its own representation especially from a `Measurement`.
+	Interface() interface{}
```

Comment:
This name seems off to me.  My first thought is that this should be `Implementation() interface{}`.  I know that it returns an `interface{}`, but it seems that the intent is to access the underlying implementation.

Reply:
:+1:

Reply:
Done.